### PR TITLE
NLDI Tweaks

### PR DESCRIPTION
**Pull request by mikejohnson51**
These are intended to address https://github.com/USGS-R/dataRetrieval/issues/546. With a few modifications. Here is a summary:

1. If only a feature is requested. The returned object is no longer a list but just the object

2. 'flowlines' are no longer returned by default unless only an origin and navigation direction is specified. Otherwise 'flowlines' **must** be included as a feature to 'find'

3. the location argument can now accept `sf` POINT objects

4. The vignette has been cleaned up to include these and @dblodgett-usgs additions

5. Documentation is improved and tests are modified

``` r
library(dataRetrieval)
library(AOI)

# No longer returned as a list if only a feature is requested
findNLDI(comid = 101)
#> Simple feature collection with 1 feature and 3 fields
#> geometry type:  LINESTRING
#> dimension:      XY
#> bbox:           xmin: -94.64845 ymin: 31.0838 xmax: -94.62997 ymax: 31.09915
#> geographic CRS: WGS 84
#> # A tibble: 1 x 4
#>   sourceName   identifier comid                                         geometry
#>   <chr>        <chr>      <chr>                                 <LINESTRING [°]>
#> 1 NHDPlus com… 101        101   (-94.64845 31.09915, -94.64803 31.09871, -94.64…
# When no find is specificed for a navigation, flowlines are returned
summary(findNLDI(comid = 101, nav = "UT"))
#>              Length Class Mode
#> origin       4      sf    list
#> UT_flowlines 2      sf    list
# Otherwise flowlines must be explicitly requested
summary(findNLDI(comid = 101, nav = "UT", find = "nwis"))
#>             Length Class Mode
#> origin      4      sf    list
#> UT_nwissite 6      sf    list

# Location by sf POINT
summary(findNLDI(location = AOI::geocode("UCSB", pt = TRUE)))
#>   sourceName         identifier           comid                    geometry
#>  Length:1           Length:1           Length:1           LINESTRING   :1  
#>  Class :character   Class :character   Class :character   epsg:4326    :0  
#>  Mode  :character   Mode  :character   Mode  :character   +proj=long...:0
```

<sup>Created on 2020-12-09 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### CODE REVIEW

File: R/findNLDI.R [hunk 1]
```diff
@@ -226,52 +240,46 @@ valid_ask = function(all, type){
...
-#'  str(findNLDI(nwis = '11120000', nav = "UM"), max.level = 1)
+#'  findNLDI(nwis = '11120000', nav = "UM")
 #'
 #' # MULTI-REQUEST
 #' # UPPER MAINSTEM and TRIBUTARY of USGS-11120000
-#'  str(findNLDI(nwis = '11120000', nav = c("UT", "UM")), max.level = 1)
+#'  findNLDI(nwis = '11120000', nav = c("UT", "UM"))
 #'
-#' # Discover Features
+#' # Discover Features(flowlines will not be returned by default)
```
Comment (dblodgett-usgs) [inline R/findNLDI.R:251]: "Flowlines will not be returned unless included in `find`" ??

File: R/findNLDI.R [hunk 2]
```diff
@@ -226,52 +240,46 @@ valid_ask = function(all, type){
 #' ## GENERAL ORIGIN: WaDE
 #'  findNLDI(origin = list("wade" = 'CA_45206'))
 #'
-#' # Navigation
+#' # Navigation (flowlines will be returned by default)
```
Comment (dblodgett-usgs) [inline R/findNLDI.R:243]: "Flowlines will be returned if `find` is not specified" ??

File: R/findNLDI.R [hunk 3]
```diff
@@ -289,53 +297,55 @@ findNLDI <- function(comid = NULL,
   }
 
   # Ensure nav types are valid
-  bad_nav = !nav %in% c("UM", "UT", "DD", "DM")
+  bad_nav <- !nav %in% c("UM", "UT", "DD", "DM")
```
Comment (dblodgett-usgs) [inline R/findNLDI.R:300]: HAH -- I was going to nag you about the = and <-

File: R/findNLDI.R [hunk 4]
```diff
@@ -344,67 +354,49 @@ findNLDI <- function(comid = NULL,
...
+                   if (getBasin) { paste0(start_url, "basin") },
+                   features)),
 
-  # define the type of each URL (feature, basin, or nav)
-  # This is needed as the attribute names for each varies
-  types = c("feature",
-            if (getBasin) { 'basin' },
-            rep("nav", length(nav)),
-            rep("feature", length(find) * length(nav)))
+    type = c("feature",
```
Comment (dblodgett-usgs) [inline R/findNLDI.R:377]: Did you mean to change this to "type" from "types" ?

**Review by dblodgett-usgs — APPROVED** [review]
A few things in here to check, but I'm ok with it.

File: R/findNLDI.R (hunk 3 above)
Comment (mikejohnson51) [reply R/findNLDI.R:300]: I knew that one was coming :P

File: R/findNLDI.R (hunk 2 above)
Comment (mikejohnson51) [reply R/findNLDI.R:243]: If this is a question, yes, findNLDI(comid = 101, nav = "UT") will currently give the upstream flowlines

File: R/findNLDI.R (hunk 4 above)
Comment (mikejohnson51) [reply R/findNLDI.R:377]: I had to re-write that chunk, I guess I just made them all singular this time haha 🤷‍♂️ . Its consistent with line 338.

File: R/findNLDI.R (hunk 2 above)
Comment (dblodgett-usgs) [reply R/findNLDI.R:243]: No -- I was suggesting different documentation.

Comment (mikejohnson51) [reply R/findNLDI.R:243]: 👍

**Comment by dblodgett-usgs** [comment]
Looks good to go from a code perspective.

File: R/findNLDI.R [hunk 5]
```diff
@@ -226,52 +240,46 @@ valid_ask = function(all, type){
...
-#' ## Convert returned list of data.frames to list of spatial features
-#' \donttest{
-#'  nldi = findNLDI(nwis = '11120000', nav = "UT", find = c("nwis", "wqp"))
-#'  str(nldi, max.level = 1)
-#'  }
+#' ## Limit search to 50 km
+#'  findNLDI(comid = 101, nav = "DM", find = c("nwis", "wqp", "flowlines"), distance_km = 50)
 
 
+location = AOI::geocode("UCSB", pt = TRUE)
```
Comment (dblodgett-usgs) [inline R/findNLDI.R:264]: Gotta drop this call to AOI. Tests are seeing it and puking.

Comment (mikejohnson51) [reply R/findNLDI.R:264]: Wow. Haha, nice catch!

**Comment by dblodgett-usgs** [comment]
Wow -- Windows latest passes but all the others fail. That's got to be some kind of first. I think you are good here. The github actions don't all have sf installed and it looks like there are some other issues on the other tiers. What do you think @ldecicco-USGS  ?

**Pull request merged**